x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/ClientTransformIndexer.java

```diff
@@ -455,20 +455,27 @@ private void injectPointInTimeIfNeeded(
         );
     }
 
-    private void doSearch(Tuple<String, SearchRequest> namedSearchRequest, ActionListener<SearchResponse> listener) {
-        logger.trace(() -> new ParameterizedMessage("searchRequest: [{}]", namedSearchRequest.v2()));
+    void doSearch(Tuple<String, SearchRequest> namedSearchRequest, ActionListener<SearchResponse> listener) {
+        SearchRequest searchRequest = namedSearchRequest.v2();
+        // We want to treat a request to search 0 indices as a request to do nothing, not a request to search all indices
+        if (searchRequest.indices().length == 0) {
+            logger.debug("[{}] Search optimized to noop; request [{}]", getJobId(), searchRequest);
```

Comment:
```suggestion
            logger.debug(() -> new ParameterizedMessage("[{}] Search optimized to noop; request [{}]", getJobId(), searchRequest));
```
Serializing the search request to string could be non trivial

Reply:
Also, it would be good to log which named search was optimized `namedSearchRequest.v1()`

Reply:
The `debug()` method is calling https://github.com/apache/logging-log4j2/blob/85b564a49f420fbcb9ac8abd3a20a252303d8198/log4j-api/src/main/java/org/apache/logging/log4j/spi/AbstractLogger.java#L449 which takes `Object` arguments.  They only have their `toString()` methods called if logging at the level is enabled.  As far as I can see the `ParameterizedMessage` approach is only necessary when you want to log _both_ a parameterized message _and_ an exception with stack trace.

Reply:
AH! Cool! Well, forget about the message supplier stuff, but having the search name would be nice.